opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -840,16 +833,18 @@ namespace Opm {
                 const double pvValue = ebosProblem.porosity(cell_idx) * ebosModel.dofTotalVolume( cell_idx );
                 pvSumLocal += pvValue;
 
-                for ( int phaseIdx = 0; phaseIdx < np; ++phaseIdx )
+                for (unsigned phaseIdx = 0; phaseIdx < FluidSystem::numPhases; ++phaseIdx)
                 {
-                    const int ebosPhaseIdx = flowPhaseToEbosPhaseIdx(phaseIdx);
-                    const int ebosCompIdx = flowPhaseToEbosCompIdx(phaseIdx);
+                    if (!FluidSystem::phaseIsActive(phaseIdx))
+                        continue;
+
+                    const unsigned compIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::solventComponentIndex(phaseIdx));
```

Comment:
Why does the solventComponentIndex enter here? That looks fishy in a general part of the code with no "if solvent" guard.

Reply:
This "solvent" is not related to the solvent model. FluidSystem::solventComponentIndex return the component Index of the solvent in the phase. i.e. gas component index for gas phase etc.

Reply:
it is "solvent" in the sense of "main component of a phase". proposals for better names are highly appreciated.

Reply:
Since the solvent model is one important feature of opm flow, it can be really confusing and problem prone for later use if we use `solvent` for something different at the same time. Can we just remove `solvent` from the `solventComponentIndex`?  Or with more detailed explanation here, maybe people can have better suggestions.

Reply:
If I remember my chemistry lessons ages ago, solvent pr def. is the substance the solute dissolves into. i.e. if gas dissolves in oil. Oil is solvent and gas is solute. The same as the naming convention in opm-material. But yes I agree this is confusing. Could primaryComponentIndex() work? Or maybe we should rename the solvent model to something more appropriate?

Reply:
> Or maybe we should rename the solvent model to something more appropriate?

It is probably the more costly solution. and it needs another `Good` name.

Reply:
Good point. Anyway changing this should not be part of this PR I think.

Reply:
> Could primaryComponentIndex() work?

yes, that will do as far as I'm concerned. I'm not perfectly happy with it either, though. will you open a PR?

(BTW: the "solvent" component of the black oil solvent extension does not seem to be a solvent in a thermodynamic sense ;))

Reply:
>> Or maybe we should rename the solvent model to something more appropriate?
>
> It is probably the more costly solution. and it needs another Good name.

I agree with @GitPaean here; even though the term "solvent" is technically incorrect in that context, the official documentation seems to use it all the time. it would thus lead to an infinite amount of confusion if we went for something else.